Readme.md

```diff
@@ -338,70 +391,125 @@ parseUser('{ "name": "john" }').get(); // { name: 'john' }
 ```ts
-sequence<A>(this: Either<Applicative<A>>,
-        of: (v: Value) => Applicative<Value>,
-    ): Applicative<Either<A>>;
-sequence<A>(
-    this: Either<Error>,
-    of: (v: Value) => Applicative<Value>,
-): Applicative<Either<Error>>;
+/*
+ * Either<Applicative<A>>.sequence(
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Either<A>>;
+ * */
+Either.of(Identity.of(5)).sequence(Identity.of);
+    -> Identity<Either<5>>
+/*
+ * Either<Error>.sequence(
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Either<Error>>;
+ * */
+Either.of(new Error('Boom')).sequence(Identity.of);
+    -> Identity<Either<new Error('Boom')>>
 ```
 - maybe.traverse: map a function returning an applicative and then swap the applicative with the either.
 If the either hold an error, an Apllicative<Maybe<Error>> will be returned
 ```ts
-traverse<A, B>(
-    this: Either<Applicative<A>>,
-    of: (v: Value) => Applicative<Value>,
-    fn: (v: A) => Applicative<B>,
-): Applicative<Either<B>>;
-sequence<A, B>(
-    this: Either<Error>,
-    of: (v: Value) => Applicative<Value>,
-    fn: (v: A) => Applicative<B>,
-): Applicative<Either<Error>>;
+/*
+ * Either<Applicative<A>>.traverse(
+ *     fn: A => Applicative<B>,
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Either<B>>;
+ * */
+Either.of(5).traverse(v => Identity.of(v * 2));
+    -> Identity<Either<10>>
+/*
+ * Either<Error>.sequence(
+ *     fn: A => Applicative<B>,
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Either<Error>>;
+ * */
+Either.of(5).traverse(v => throw new Error('Boom'));
+    -> Identity<Either<new Error('Boom')>>
 ```
 - maybe.catch: works like map, it accept a function, and will call it if the value is an error. It Will transform the Error in a value an return a Right Error.
 ```ts
-catch<A, B>(this: Either<A>, fn: (error: Error) => B): Either<A>;
-catch<A, B>(this: Either<Error>, fn: (error: Error) => B): Either<B>;
+// Either<A>.catch(fn: Error => B): Either<A>;
+Either.of(5).catch(error => error.message);
+    -> Either<5>
+// Either<Error>.catch(fn: Error => B): Either<B>;
+Either.of(new Error('Boom')).catch(error => error.message);
+    -> Either<'Boom'>
 ```
 - maybe.isLeft: return true if the maybe holds an error
+```ts
+Either<A>.isLeft(): false;
+Either<Error>.isLeft(): true;
+```
 - maybe.isRight: return true if the maybe holds a value
+```ts
+Either<A>.isRight(): true;
+Either<Error>.isRight(): false;
+```
 - maybe.get: return the value or throw the error
+```ts
+Either<A>.get(): A;
+Either<Error>.get(): throw Error;
+```
 
 ## IO
 
 IO is a shorthand for `Input/Output` IO is used to handle side effect. Instead of holding the value directly, we hold a function that will return the value. The important distinction compared to the previous monads is that IO is lazy. We can map all we want, nothing will happen until you call its execute method.
```

Comment:
side effects
IO holds
lazy: the map function can be called several times, nothing will be executed until the `execute` method is called